R/readWriteMgfData.R

```diff
@@ -20,13 +20,28 @@ setMethod("writeMgfData",
                              verbose = verbose)
           })
 
+#' @param addFields `data.frame` or `matrix` with optional additional
+#'     fields to be added to each spectrum (each column one field).
+#'
+#' @noRd
 writeMgfDataFile <- function(splist, con, COM = NULL, TITLE = NULL,
-                             verbose = isMSnbaseVerbose()) {
+                             verbose = isMSnbaseVerbose(), addFields = NULL) {
   if (class(con) == "character" && file.exists(con)) {
     message("Overwriting ", con, "!")
     unlink(con)
   }
 
+  if (length(addFields)) {
+      if (length(dim(addFields)) != 2)
+          stop("'addFields' has to be a matrix or data.frame.")
+      if (!is.matrix(addFields))
+          addFields <- do.call(cbind, lapply(addFields, as.character))
```

Comment:
I really hate `as.matrix` for calling `format` and padding all `character`s with whitespaces. Instead of `do.call` and `lapply` we could use `addFields <- trimws(as.matrix(d))` here.

Reply:
BTW: Internal `as.matrix.data.frame` uses a `for` loop. So depends on your style whether you like the `do.call, cbind, lapply` or `trimws(as.matrix())` approach.

Reply:
I'd rather take the `do.call, cbind, lapply` avenue - it's faster then the `trimws(as.matrix)`. But thanks for the suggestion.